lib/screens/pets/pet-profile/pet_profile.dart

```diff
@@ -0,0 +1,316 @@
+                                if (displayedPet.forPetMating)
+                                  FlatButton(
+                                    onPressed: () => {},
+                                    padding: EdgeInsets.all(10),
+                                    color: Colors.brown,
+                                    child: Text(
+                                      "Mating",
+                                      style: TextStyle(color: Colors.green),
+                                    ),
+                                  )
+                              ],
+                            )
+                          : Row(
+                              mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                              children: <Widget>[
+                                if (displayedPet.forPetSitting)
+                                  FlatButton(
+                                    onPressed: () => {
+                                      Navigator.of(context).pushNamed(
+                                          AddEditPetPage.routeName,
+                                          arguments: displayedPet)
+                                    },
+                                    padding: EdgeInsets.all(10),
+                                    color: Colors.brown,
+                                    child: Text(
+                                      "Edit",
+                                      style:
+                                          TextStyle(color: Colors.amberAccent),
+                                    ),
+                                  ),
+                                if (displayedPet.forPetMating)
+                                  FlatButton(
+                                    onPressed: () => {
+                                      showDialog(
+                                          context: context,
+                                          builder: (context) {
+                                            return confirmDeleteDialog(context);
+                                          })
+                                    },
+                                    padding: EdgeInsets.all(10),
+                                    color: Colors.brown,
+                                    child: Text(
+                                      "Delete",
+                                      style: TextStyle(color: Colors.green),
+                                    ),
+                                  )
+                              ],
+                            )
+                    ],
+                  ),
+                ),
+              )
+            ],
+          ),
+        )));
+  }
+
+  deletePet(BuildContext context, String petId) async {
+    final PetsService petsService = services.get<PetsService>();
+    await petsService.deletePet(petId);
+
+    // to be changed. It works partially - back button to previous pet list still works...
```

Comment:
What is the desired behaviour?
Is `popUntil` what you wanted to do?

Reply:
Changed